spaceship/lib/spaceship/tunes/iap.rb

```diff
@@ -75,6 +75,17 @@ def create!(type: "consumable",
                            family_id: family_id,
                            subscription_duration: subscription_duration,
                            subscription_free_trial: subscription_free_trial)
+
+        # Update pricing for a recurring subscription.
+        if type == Spaceship::Tunes::IAPType::RECURRING && pricing_intervals
+          # There are cases where the product that was just created is not immediately found,
+          # and in order to update its pricing the purchase_id is needed. Therefore polling is done
+          # until it is found.
+          product = find(product_id) until product
+          client.update_recurring_iap_pricing!(app_id: self.application.apple_id,
```

Comment:
btw, I found one little bug here, pricing_intervals passed in this method should be the full pricing intervals for all countries. This list is generated in `iap_detail.rb` before saving (here https://github.com/fastlane/fastlane/pull/11627/files#diff-0ad09501b52f13ed83e0709d8a5b1df3L180)

Reply:
You're right, I fixed it. For now I copy the code from `iap_detail.rb`. In the future I suggest to have one place that does this transformation, I think `tunes_client.rb` is a good place for this.   It is already done in `create_iap!` method.